lib/senders/index.js

```diff
@@ -89,12 +123,11 @@ module.exports = function (config, log) {
             uaBrowserVersion: opts.uaBrowserVersion,
             uaOS: opts.uaOS,
             uaOSVersion: opts.uaOSVersion
-          }
-        ))
-      }
-      mailer.sendPasswordChangedNotification = function (email, opts) {
-        return P.resolve(mailer.passwordChangedEmail(
-          {
+          }))
+      },
+      sendPasswordChangedNotification(email, opts) {
+        return bounceGatedMailer(email)
+          .then(mailer => mailer.passwordChangedEmail({
```

Comment:
Hmmm...this is an interesting edge-case that I don't recall us discussing.  This will suppress sending of notification emails like "a new device was added to your account" or "your password was changed".  If you have a hard bounce on your email, what will happen here IIUC is:

* You (or an attacker!) attempt to change the account password
* We store the updated password in the database
* We try to send a notification email, but fail due to this gating
* We report an error saying we were unable to chance your password because your email bounces.

But, while the request failed, the actual changing of the password will have succeeded!  Are there bad UX or security implications to this?

/cc @davismtl for input